### PR TITLE
add test 3: connectivity to all endpoints of service imported by multiple clusters

### PR DESCRIPTION
**Pull request by nojnhuh**
This implements test 3 from [KEP 1645](https://github.com/kubernetes/enhancements/blob/master/keps/sig-multicluster/1645-multi-cluster-services-api/README.md#test-plan). The new test is based on the existing connectivity test, exporting the service from cluster1 in addition to cluster2 and verifying pods in each cluster are reachable.

### CODE REVIEW

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Comment by nojnhuh** [comment]
> Is there a reason why we aren't running these tests routinely on PRs to mcs-api:master ?

I see a dashboard for the SIG but there's nothing in it currently: https://testgrid.k8s.io/sig-multicluster

I don't even see a sig-multicluster directory anywhere so I can try at least creating that with an OWNERS file as long as that doesn't require any special permissions until approval. Once we have that bootstrapped I can try adding the presubmit job definition.

**Comment by nojnhuh** [comment]
> > Is there a reason why we aren't running these tests routinely on PRs to mcs-api:master ?
> 
> I see a dashboard for the SIG but there's nothing in it currently: https://testgrid.k8s.io/sig-multicluster
> 
> I don't even see a sig-multicluster directory anywhere so I can try at least creating that with an OWNERS file as long as that doesn't require any special permissions until approval. Once we have that bootstrapped I can try adding the presubmit job definition.

https://github.com/kubernetes/test-infra/pull/33369

FYI @lauralorenz @skitt @JeremyOT

**Comment by jackfrancis** [comment]
> https://github.com/kubernetes/test-infra/pull/33369

Love to see it

File: e2etest/connectivity_two_cluster_test.go
Comment (tpantelis) [inline e2etest/connectivity_two_cluster_test.go:33]: Since this is a different test than the existing one, the textual description should differ.

File: e2etest/connectivity_two_cluster_test.go
Comment (tpantelis) [inline e2etest/connectivity_two_cluster_test.go:155]: There is a lot of code duplicated with _connectivity_test.go_.  It looks like much of it could be reused. The vars above on L37-40 could be moved to a struct and the common code moved to methods on the struct.

Comment (nojnhuh) [reply e2etest/connectivity_two_cluster_test.go]: I refactored this to slot these new tests into the existing connectivity test definition.

Comment (tpantelis) [reply e2etest/connectivity_two_cluster_test.go]: 👍

**Comment by skitt** [comment]
/approve

@jackfrancis could you lgtm again?

**Pull request merged**
